apollo-router/src/plugins/telemetry/mod.rs

```diff
@@ -105,6 +110,7 @@ impl Drop for Telemetry {
                         opentelemetry::trace::TracerProvider::force_flush(&tracer_provider);
                     });
                     futures::executor::block_on(jh).expect("failed to flush tracer provider");
+                    global::shutdown_tracer_provider();
```

Comment:
I think this block of code from 106-115 may be more complicated than we need. If the drop always occurs within the context of a tokio runtime (which I think it should do), then we don't need to spawn a thread and create a new runtime, but we do need to put the `shutdown_tracer_provider() call into the spawn_blocking() (to make sure it runs on a different thread.

Reply:
Wow, I actually added ` global::shutdown_tracer_provider();` during testing and forgot to remove it.
This would have broken tracing on reload. I'm going to add something to the integration test to force a config reload and try again.
Regarding the complexity of the code above, I'll have another go once the test is in place.

Reply:
You were right. No need to have a new runtime. This was super easy to test with the new hot reload integration test.